### ISSUE TITLE
coremlc: Error: compiler error:  Invalid data blob: 'cnn_net_1/conv-maxpool-5/pool_5:0' shape (C,H,W = 150,146,-144) for output of layer: 'cnn_net_1/conv-maxpool-5/pool_5:0'.

### ISSUE THREAD

**Issue by gxdalu-yaya** [tf-coreml/tf-coreml#79]
after i converted the tfmodel to mlmodel, when i run the predict program, the following errors appeared:
coremlc: Error: compiler error:  Invalid data blob: 'cnn_net_1/conv-maxpool-5/pool_5:0' shape (C,H,W = 150,146,-144) for output of layer: 'cnn_net_1/conv-maxpool-5/pool_5:0'.

in fact, it's a max_pooling operation, and the previous layers's shape is （C,H,W）= (150, 146, 1), the operation is ：
pooled = tf.nn.max_pool(
                        conv_h,
                        ksize=[1, self.sequence_length - filter_size + 1, 1, 1], 
                        strides=[1, 1, 1, 1], 
                        padding='VALID',
                        name="pool_%s" % filter_size)

how could i solve the problem? thank you

**Comment by slin07** [comment]
Can you provide a little more information, so we can try to reproduce the result:
(1) What model are you trying to convert?
(2) What are the arguments you use for the converter?
Thanks!

**Comment by aseemw** [comment]
@gxdalu-yaya It seems that you are doing 1D convolution and there seems to be a mismatch in the axes on which the input is mapped to in CoreML (C,H) and the axes on which the kernel filters are mapped (C,W). If you share the model or the code you used to build it, we will be able the debug the issue.

**Comment by azamsharp** [comment]
AseemW: I think it would be a great idea for Google to either write an article or create a 5 min video of how to do the conversion using tfcoreml tools. Take the "Inception" model and show us step by step how to convert it to Core ML model. The model should take image as input and give a prediction result as output indicating the confidence in different items it came up with for the picture. 

I think tfcoreml is a great tool but with no documentation and no way to know how to use it.. it becomes useless.

**Comment by aseemw** [comment]
@azamsharp I am sorry to hear that you are having issues using the class label file. 
The python notebook describes the process of converting inception to CoreML and takes image as an input: https://github.com/tf-coreml/tf-coreml/blob/master/examples/inception_v3.ipynb
I agree, it is a good idea to add the usage of "classLabel" argument to that example. 

PR#86 fixed an issue we had in the classLabel argument. We are looking into issue #89

**Comment by azamsharp** [comment]
I used the following to install the tfcoreml tools. 

pip install -U tfcoreml

How can I get the updated version which has the class_labels fix?

**Comment by azamsharp** [comment]
I downloaded the latest version of tfcoreml tools (tensorflow-1.4.1) and I have the same exact issue. 

<img width="880" alt="screen shot 2017-12-15 at 3 16 35 pm" src="https://user-images.githubusercontent.com/16604/34060822-08c33164-e1ab-11e7-81e9-0ecf303c5481.png">

**Comment by aseemw** [comment]
Kindly see issue #89. That is the correct thread to discuss this issue.

**Comment by aseemw** [comment]
@gxdalu-yaya I am closing this issue now. Please reopen another one when you provide more details on the architecture that you are trying to convert.
